src/repr/src/scalar.rs

```diff
@@ -881,6 +899,11 @@ pub enum ScalarType {
     TimestampTz,
     /// The type of [`Datum::Interval`].
     Interval,
+    /// A single byte character type backed by a [`Datum::UInt8`].
+    ///
+    /// PostgreSQL calls this type `"char"`. Note the quotes, which distinguish
+    /// it from the type `ScalarType::Char`.
+    Byte,
```

Comment:
Nitpick: since we also have a `Bytes`, could we maybe call this `SingleByte` or rename `Bytes` to `ByteArray` or something else?
Having `Byte` and `Bytes` seems to just add more confusion to me, especially since from the name I'd expect them to be something more similar to `char` / `varchar` and `bit` / `varbit` (which we still don't have, but probably will have eventually) or `binary` / `varbinary` (which is the MySQL / SQL Server approach to storing non-blob binary data).

Reply:
Ah, fair enough. I renamed this to `ScalarType::PgLegacyChar`. Figure we might as well discourage anyone from thinking that it is a good idea to use it for anything.